From 047fe525003acb4298345f7ea0b03ea74f11c2a8 Mon Sep 17 00:00:00 2001
From: Gerd Hoffmann <kraxel@redhat.com>
Date: Thu, 29 Jan 2026 10:45:28 +0100
Subject: [PATCH 32/32] MdeModulePkg/CapsuleOnDiskLoadPei: fix gcc 16 warning
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

MdeModulePkg/Universal/CapsuleOnDiskLoadPei/CapsuleOnDiskLoadPei.c:176:11: error: variable ‘Index’ set but not used [-Werror=unused-but-set-variable=]
  176 |   UINTN   Index;
      |           ^~~~~

Signed-off-by: Gerd Hoffmann <kraxel@redhat.com>
---
 .../Universal/CapsuleOnDiskLoadPei/CapsuleOnDiskLoadPei.c      | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/MdeModulePkg/Universal/CapsuleOnDiskLoadPei/CapsuleOnDiskLoadPei.c b/MdeModulePkg/Universal/CapsuleOnDiskLoadPei/CapsuleOnDiskLoadPei.c
index c7be6c52525c..4d4f5fde2f6f 100644
--- a/MdeModulePkg/Universal/CapsuleOnDiskLoadPei/CapsuleOnDiskLoadPei.c
+++ b/MdeModulePkg/Universal/CapsuleOnDiskLoadPei/CapsuleOnDiskLoadPei.c
@@ -173,7 +173,6 @@ RetrieveRelocatedCapsule (
   IN UINTN  RelocCapsuleTotalSize
   )
 {
-  UINTN   Index;
   UINT8   *CapsuleDataBufEnd;
   UINT8   *CapsulePtr;
   UINT32  CapsuleSize;
@@ -244,7 +243,6 @@ RetrieveRelocatedCapsule (
   // Re-iterate the capsule buffer to create Capsule hob & Capsule Name Str Hob for each Capsule saved in relocated capsule file
   //
   CapsulePtr = RelocCapsuleBuf;
-  Index      = 0;
   while (CapsulePtr < CapsuleDataBufEnd) {
     CapsuleSize = ((EFI_CAPSULE_HEADER *)CapsulePtr)->CapsuleImageSize;
     BuildCvHob ((EFI_PHYSICAL_ADDRESS)(UINTN)CapsulePtr, CapsuleSize);
@@ -252,7 +250,6 @@ RetrieveRelocatedCapsule (
     DEBUG ((DEBUG_INFO, "Capsule saved in address %x size %x\n", CapsulePtr, CapsuleSize));
 
     CapsulePtr += CapsuleSize;
-    Index++;
   }
 
   return EFI_SUCCESS;
-- 
2.52.0

